### ISSUE TITLE
[Bug] Memory consumed increases and rises 

### ISSUE THREAD

**Issue by wurmrobert** [toni-moreno/snmpcollector#372]
- SnmpCollector Version 0.7.7
- Docker Container with alpine:latest on a ubuntu machine

I am using your collector for about 550 devices.

Each Device has:

- 4 indexed with direct tag measuerments
- 1 measurement group
- no custom filters 

and this settings:

- timeout: 20
- retries: 5
- log level: info
- snmpdebug: false
- snmp version: 2c
- disable bulk: false
- max repetitions: 5
- freq: 30
- update flt freq: 60
- concurrent gather: true

The memory reservation of the instance after one day:
<img width="794" alt="screen shot 2018-09-26 at 23 27 49" src="https://user-images.githubusercontent.com/9077841/46129383-c2fa3480-c236-11e8-9e51-77677a47b2b2.png">

The gvm dashboard after some hours:
<img width="980" alt="screen shot 2018-09-27 at 08 52 35" src="https://user-images.githubusercontent.com/9077841/46129396-d1e0e700-c236-11e8-9f76-aa5a4cccc3cd.png">

Here some generated profiles:
[profiles.zip](https://github.com/toni-moreno/snmpcollector/files/2422901/profiles.zip)

Have you ever tried to run the collector for this number of devices?
Thanks in advance.

**Comment by wurmrobert** [comment]
Does someone has the same behavior?d

**Comment by toni-moreno** [comment]
Hi @wurmrobert sorry for the late response. I will review ASAP this issue.

**Comment by toni-moreno** [comment]
Hi @wurmrobert , in next days we will release 0.8 version with some new features , and fixes that could improve memory management. If not , the new release will add a new self metric statistic "buffer_percent_used",  I suspect that memory consumption has to be with the use of the output buffer that snmpcollector could use.  
(be careful with the upgrade  , you should update metric table by hand, following the CHANGELOG instructions)

**Comment by wurmrobert** [comment]
Hi. Thanks a lot. After the release i will let you know about the status.

**Comment by wurmrobert** [comment]
Hi @toni-moreno .
Can i find the changes of the memory management already in a feature branch, so that i can test it?

**Comment by toni-moreno** [comment]
@wurmrobert  I've released a new 0.8 version with the main hability to add , delete devices online without downtime, but it has also  new internal [statistics](https://github.com/toni-moreno/snmpcollector/wiki/Troubleshooting#selfmon_outdb_stats)  to get "buffer_percend_used"  buffering status. That in my opinion could be related with this memory issue. Could you also attach new profile data please?

  Be carefull with  the [breaking changes](https://github.com/toni-moreno/snmpcollector/blob/master/CHANGELOG.md#breaking-changes)

**Comment by wurmrobert** [comment]
Thanks a lot @toni-moreno for your work!

I have tested and played around the last days a lot. I have increased the device freq and updated golang from v0.9 to v1.1.0. Also i have decreased the buffersize from the the default value 65535 to 2000. After this everything works well. The number of goroutines and used memory is as expected:

<img width="1296" alt="screenshot 2018-11-06 at 08 59 45" src="https://user-images.githubusercontent.com/9077841/48050677-cd4b0f00-e1a2-11e8-905a-f0fdd06724a8.png">


Currently i am looking what what was the key for the bug. I will update to your latest version in the next days and will let you know about the status and the statistics.
